aws-s3/scripts-ccao-data-raw-us-east-1/spatial-airport_noise_point_source_new_models.R

```diff
@@ -0,0 +1,357 @@
+                                                        BASELINE_DNL)
+
+# Verify against https://www.omnicalculator.com/physics/db
+
+## PREDICT AIRPORT NOISE LEVELS ================================================
+
+# Set intercept at 0, i.e. when linear distance from airport is infinite,
+# the absolute intensity of sound emanating from it will be 0
+# (Decibel measure will NOT drop to 0; it can keep going to -Inf since 0 dB is
+# just a cutoff of human *perception* of sound)
+
+ohare_model0_intensity <- lm(added_intensity ~ 0 + isq_dist_ohare,
+                             data = airport_clean %>%
+                               filter(airport == "ohare"))
+# Adjusted R-squared:  0.4002
+
+midway_model0_intensity <- lm(added_intensity ~ 0 + isq_dist_midway,
+                              data = airport_clean %>%
+                                filter(airport == "midway"))
+# Adjusted R-squared:  0.8519
+
+# for future reference:
+model0_coefficients <- cbind(
+  airport = c("ohare", "midway"),
+  coefficient = c(
+    ohare_model_intensity$coefficients,
+    midway_model_intensity$coefficients
+  )
+)
+row.names(model0_coefficients) <- NULL
+
+# To get predicted DNL level for a PIN:
+# for each airport:
+# -take the PIN's distance to that airport to the -2nd power (=1/(d^2))
+# -apply coefficient to that inverse-square distance
+# -convert result to decibels to get estimated decibels added by that airport
+# -if decibel level is below 0, use 0 instead
+# -add O'Hare and Midway results together
+# -add BASELINE_DNL (50) to that sum
+
+# ==============================================================================
+# MODEL 1: most recent values
+# (O'Hare Modernization Project levels for O'Hare; 2019 levels for Midway)
+
+ohare_recent <- ohare_noise %>%
+  select(-c(noise, year)) %>%
+  distinct() %>%
+  rename("noise" = "modeled_omp_build_out_values") %>%
+  filter(!(is.na(noise))) # four sensors have NA values for OMP; removed those
+
+midway_recent <- midway_noise %>%
+  select(-modeled_omp_build_out_values) %>%
+  filter(year == 2019) %>%
+  select(-year)
+
+airport_recent <- rbind(ohare_recent, midway_recent)
+
+sensor_dists1 <- st_distance(airport_recent, aps)
+
+airport_recent$distance_ohare <- sensor_dists1[, 1]
+airport_recent$distance_midway <- sensor_dists1[, 2]
+
+airport_recent$isq_dist_ohare <- (airport_recent$distance_ohare)^-2
+airport_recent$isq_dist_midway <- (airport_recent$distance_midway)^-2
+
+airport_recent$total_intensity <- decibel_to_intensity(airport_recent$noise)
+airport_recent$added_intensity <- decibel_to_intensity(airport_recent$noise -
+                                                         BASELINE_DNL)
+
+ohare_model1_intensity <- lm(added_intensity ~ 0 + isq_dist_ohare,
+                             data = airport_recent %>%
+                               filter(airport == "ohare"))
+summary(ohare_model1_intensity)
+# Adjusted R-squared: 0.6612
```

Comment:
**thought:** Looks like we have a winner. It also seems most defensible to me to use the latest values from each airport and consider them steady into the future. I say we go with this model. What do you think?

Reply:
Choosing values from a single year also makes the resulting feature easier to update since we only have to change the resulting model coefficients if we get new monitor values.